### DISCUSSION TITLE
Strategy for Supabase Auth

### DISCUSSION THREAD

**Discussion by lucascurti** [sergiodxa/remix-auth#5] [Q&A] [answered]
Thinking about using [Supabase Authentication](https://supabase.io/docs/reference/javascript/auth-signin) for a Remix app I'm building. I was wondering if it's something that `remix-auth` supports now and what Strategy should be used. I haven't played with it yet but I think that I can do something with the Local Strategy to make it work. Would this be the right approach?

Edit: Nevermind. I've just found the Custom Strategy. I'll give it a try and report back.

**Comment by sergiodxa** [comment, marked as answer]
The CustomStrategy is great in case you want to do anything you want without writing the strategy class yourself, but if you do write a strategy class you can externalize it and test it in isolation of your application (that’s the reason this package exists).

Now talking specifically about Supabase, if they support server side authentication it would be possible to use it with Remix Auth, since they support auth with third party providers through OAuth2 it will be great if they have the required endpoints to use them with OAuth2 similar to Auth0, this way you will be able use email + password, magic link or third party providers.

**Comment by koriner** [comment]
@lucascurti Did you put together a working example of this at all? I'm thinking about doing the same thing with Supabase auth, if you feel like sharing your implementation that would be awesome!

**Reply by lucascurti** [reply]
Hey @koriner. I didn't. Supabase doesn't provide OAuth2 endpoints and I ended up using my own implementation with user & password without Supabase. Sergio asked Supabase for adding those endpoints but I don't think they answered to that one yet.

**Comment by MichaelDeBoey** [comment]
@lucascurti Thanks to @mitchelvanbever, [`remix-auth-supabase-strategy`](https://github.com/mitchelvanbever/remix-auth-supabase-strategy) now exists 🥳🎉
